### PR TITLE
refactor options pkg

### PR DESCRIPTION
**Pull request by andyxning**
Refactor `options` package to `cmd` directory.

/cc @Random-Liu @dchen1107

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/node-problem-detector/96)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -81,7 +81,7 @@ version:
 ./bin/node-problem-detector: $(PKG_SOURCES)
 	CGO_ENABLED=$(CGO_ENABLED) GOOS=linux go build -o bin/node-problem-detector \
 	     -ldflags '-X $(PKG)/pkg/version.version=$(VERSION)' \
-	     $(BUILD_TAGS) cmd/node_problem_detector.go
+	     $(BUILD_TAGS) cmd/node_problem_detector/node_problem_detector.go
```
Comment (Random-Liu) [inline Makefile]: Because we only have one binary, I'd prefer organize it like this for now:
```
cmd/node_problem_detector.go
cmd/options/options.go
```

**Comment by Random-Liu** [comment]
LGTM with one nit.

**Comment by andyxning** [comment]
@Random-Liu Comments addressed. PTAL.

**Comment by Random-Liu** [comment]
LGTM

**Pull request merged**